src/Sentry.Unity/ScriptableSentryUnityOptions.cs

```diff
@@ -135,7 +135,7 @@ internal SentryUnityOptions ToSentryUnityOptions(bool isBuilding, ISentryUnityIn
     {
         application ??= ApplicationAdapter.Instance;
 
-        var options = new SentryUnityOptions(isBuilding, application)
+        var options = new SentryUnityOptions(isBuilding, application, unityInfo)
```

Comment:
We need to pass the `UnityInfo` into the options if we want to have access to `IL2CPP` from within the integrations.
Alternatively, I'd need to add the integrations here but this would be a breaking change in behaviour for anyone that programmatically initializes the SDK.

I'll have to pick this back up when working on https://github.com/getsentry/sentry-unity/issues/2200